src/DynamoCore/Engine/EngineController.cs

```diff
@@ -44,6 +45,8 @@ public class EngineController : LogSourceBase, IAstNodeContainer, IDisposable
         /// </summary>
         internal static event Action VMLibrariesReset;
 
+        internal Version WorkspaceVersion { get; set; }
```

Comment:
the name of this property confused me at first glance, shouldn't this be called something like engine version? I would think that WorkspaceVersion is the version of the currently loaded workspace.

Reply:
I moved this property from Workspace to EngineController as part of this latest refactoring as I need it before the workspace is initialized. It's meant to be the Dynamo version in which the workspace was initially created.

Reply:
Okay, I guess my confusion stems from the fact that I see it get set from the AssemblyHelper? Maybe on EngineController it could have a different name?

Reply:
It still gets serialized with the workspace - correct?

Reply:
It gets set from AssemblyHelper only if there's no version serialized with the workspace being read. Yes, it still gets serialized as part of `DynamoPreferencesData`. I can think of another name.